support-lib/jni/Marshal.hpp

```diff
@@ -80,7 +80,11 @@ namespace djinni
 		Bool() : Primitive("java/lang/Boolean", "valueOf", "(Z)Ljava/lang/Boolean;", "booleanValue", "()Z") {}
 		friend JniClass<Bool>;
 		friend Primitive<Bool, bool, jboolean>;
-		static JniType unbox(JNIEnv* jniEnv, jmethodID method, jobject j) noexcept { return jniEnv->CallBooleanMethod(j, method); }
+		static JniType unbox(JNIEnv* jniEnv, jmethodID method, jobject j) noexcept {
+            const auto result = jniEnv->CallBooleanMethod(j, method);
+            jniExceptionCheck(jniEnv);
```

Comment:
This almost makes me think there should be templated wrappers in the support lib for calling into JNI methods and always combine call + exception check as that pattern is now ubiquitous.